### ISSUE TITLE
Improve efficiency with larger data beyond h2o_data

### ISSUE THREAD

**Issue by linjing-lab** [h2oai/h2o-3#15919]
**Is your feature request related to a problem? Please describe.**
I found the released Python software named h2o contains a folder with 2 csv data, which named [h2o_data](https://github.com/h2oai/h2o-3/tree/master/h2o-py/h2o/h2o_data), and the maximum number of sample rows in the both is up to 381, far smaller than the [bitcoin dataset](https://github.com/linjing-lab/easy-pytorch/blob/main/released_box/data/bitcoin_heist_data.csv) I used when develop [perming](https://github.com/linjing-lab/easy-pytorch/tree/main/released_box), the whole samples of what I adopt is up to 1 million rows and 29 categories, and the data training and validation for each batch (batch_size=256) only took less than 0.2 seconds when cuda backend was enabled in the training cycle of perming, though I trained bitcoin dataset on NVIDIA GeForce MX250, not a particularly high-speed chip.

**Describe the solution you'd like**
I thought h2o could take cupy as a important dependency to accelerate the efficiency of larger data by adopting advanced encapsulation on the cuda side while balancing CPU parallel efficiency in numpy.

**Describe alternatives you've considered**
A perceptron algorithm without considering multiple label inputs can greatly improve training efficiency for the same machine learning task, like the perceptron model used by [perming](https://pypi.org/project/perming/), it has trained over 1 million rows of sample inputs within a 2GB cuda cache.

**Additional context**
Parallel efficiency is determined by the number of chip cores, I don't think a [wrapped estimator of scikit-learn](https://github.com/h2oai/h2o-3/blob/master/h2o-py/h2o/sklearn/wrapper.py) could bring significant efficiency improvements in runtime collaboration between training and deployment when encounter large dataset with high-dimensional features.

**Comment by mn-mikke** [comment]
@linjing-lab H2O runtime doesn't utilise GPUs for training models. The only exception is the XGBoost algorithm, but the support there is also limited. It can utilise GPUs only on a single-node H2O cluster. Currently, there is no plan to extend GPU support in H2O-3 project.

**Comment by linjing-lab** [comment]
I found xgboost.py from h2o/estimators, and it was compatibled with XGBoost algorithm by configuring backend and gpu_id, a GPU is used if available when backend='auto'. If the support is limited, I think download [xgboost from dmlc](https://github.com/dmlc/xgboost) could collaborate with advanced encapsulation algorithms in h2o while h2o enables a feasible class for users to adjust.

**Comment by mn-mikke** [comment]
H2O internally use a modified version of the xgboost library from dmlc, but GPU support for multi-node H2O clusters is not a priority for us right now.

**Comment by linjing-lab** [comment]
alright, I just pointed out that the data samples in h2o_data is relatively small, not asking h2o to support GPU parallelism as a priority while adopt multi-node h2o clusters.
